### PR TITLE
Delete users after GRACE_PERIOD if they're already disabled Users that have not logged in within GRACE_PERIOD and are going to be disabled will get immediately deleted

### PR DESCRIPTION
**Pull request by gdestuynder**
@jdow please test this with NOOP and so on first as this is very dangerous.
double check everything

### CODE REVIEW

File: DuoSecurity/duo_client_purge.py [hunk 1]
```diff
@@ -64,3 +82,13 @@ def get_ldap_user_by_mail(ldap_conn, email):
...
             if NOOP == False:
                 admin_api.update_user(duo_user_id, status='disabled')
+
+    # Delete users that have been disabled and haven't logged in for GRACE_PERIOD seconds (licenses)
+    for duo_entry in all_disabled_duo_usernames:
+        last_login = duo_entry.get('last_login')
+        if last_login:
+            unix_last_login = int(last_login)
+            if int(time.time()) - unix_last_login > GRACE_PERIOD:
+                print("Deleting user {} ID {}".format(duo_entry.get('email_address'), duo_entry.get('id')))
```
Comment (jdow) [inline DuoSecurity/duo_client_purge.py]: python 2.6.6 complains about ValueError: zero length field name in format here. I believe simply numbering the substitutions like ..{0}..{1}.. solves this.

File: DuoSecurity/duo_client_purge.py [hunk 2]
```diff
@@ -64,3 +82,13 @@ def get_ldap_user_by_mail(ldap_conn, email):
                   % (duo_email, duo_user_id)
             if NOOP == False:
                 admin_api.update_user(duo_user_id, status='disabled')
+
+    # Delete users that have been disabled and haven't logged in for GRACE_PERIOD seconds (licenses)
+    for duo_entry in all_disabled_duo_usernames:
+        last_login = duo_entry.get('last_login')
+        if last_login:
+            unix_last_login = int(last_login)
+            if int(time.time()) - unix_last_login > GRACE_PERIOD:
```
Comment (jdow) [inline DuoSecurity/duo_client_purge.py]: Need to "import time" before this works on python 2.6.6

File: DuoSecurity/local_settings.py-dist [hunk 3]
```diff
@@ -1,6 +1,7 @@
 LDAP_HOST = '127.0.0.1'
 LDAP_USERNAME = ''
 LDAP_PASSWORD = ''
+GRACE_PERIOD = 10080
```
Comment (jdow) [inline DuoSecurity/local_settings.py-dist]: Depending on how strapped for licenses we are, I suggest starting this out at more like 30 or 60 or even 90 days, as we might want to hold onto login history for a given user for forensic purposes longer than 7 days

File: DuoSecurity/duo_client_purge.py (hunk 1 above)
Comment (gdestuynder) [reply DuoSecurity/duo_client_purge.py]: oh my..
wish we had ci on this hehe

File: DuoSecurity/duo_client_purge.py (hunk 2 above)
Comment (gdestuynder) [reply DuoSecurity/duo_client_purge.py]: done

**Comment by gdestuynder** [comment]
@jdow

**Comment by gdestuynder** [comment]
@jdow

**Comment by gdestuynder** [comment]
r+ by email from jdow

**Pull request merged**